go-controller/pkg/ovn/pods.go

```diff
@@ -321,38 +329,62 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 			}
 		}
 
-		addresses = podMac.String() + " " + util.JoinIPNetIPs(podIfAddrs, " ")
-		args = []string{
-			"--may-exist", "lsp-add", logicalSwitch, portName,
-			"--", "lsp-set-addresses", portName, addresses,
-		}
 	}
-	args = append(args, "--", "set", "logical_switch_port", portName, "external-ids:namespace="+pod.Namespace, "external-ids:pod=true")
-	args = append(args, "--", "lsp-set-port-security", portName, addresses)
 
-	out, stderr, err = util.RunOVNNbctl(args...)
+	// set addresses on the port
+	addresses = make([]string, len(podIfAddrs)+1)
+	addresses[0] = podMac.String()
+	for idx, podIfAddr := range podIfAddrs {
+		addresses[idx+1] = podIfAddr.IP.String()
+	}
+
+	lspSetAddrCmd, err := oc.ovnNBClient.LSPSetAddress(portName, addresses...)
 	if err != nil {
-		return fmt.Errorf("error while creating logical port %s stdout: %q, stderr: %q (%v)",
-			portName, out, stderr, err)
+		return fmt.Errorf("unable to create LSPSetAddress command for port: %s", portName)
+	}
+	cmds = append(cmds, lspSetAddrCmd)
+
+	// add external ids
+	extIds := map[string]string{"namespace": pod.Namespace, "pod": "true"}
+
+	lspSetExtIdsCmd, err := oc.ovnNBClient.LSPSetExternalIds(portName, extIds)
+	if err != nil {
+		return fmt.Errorf("unable to create LSPSetAddress command for port: %s", portName)
+	}
+
```

Comment:
@abhat remove this empty line? at least we should be consistent about whether there's an empty line between the } of the error condition and the append...